File: OpenSilver.Samples.TelerikUI/OpenSilver.Samples.TelerikUI/Samples/Controls/RadChart3D/Extensions/ExtendedCameraExtension.vb
Imports Telerik.Windows.Controls.Charting

Namespace OpenSilver.Samples.TelerikUI.RadChart3D.Extensions
    Public Class ExtendedCameraExtension
        Inherits CameraExtension
        Public Overrides Sub Attach(owner As ChartArea)
            MyBase.Attach(owner)
            AddHandler owner.MouseWheel, AddressOf OnMouseWheel
        End Sub

        Public Overrides Sub Detach(owner As ChartArea)
            MyBase.Detach(owner)
            RemoveHandler owner.MouseWheel, AddressOf OnMouseWheel
        End Sub


        Private Sub OnMouseWheel(sender As Object, e As Windows.Input.MouseWheelEventArgs)
            If Not ZoomEnabled Then
                Return
            End If

            If e.Delta > 0 Then
                Zoom(1.1)
            Else
                Zoom(0.9)
            End If

            e.Handled = True
        End Sub
    End Class
End Namespace

File: OpenSilver.Samples.TelerikUI/OpenSilver.Samples.TelerikUI/Samples/Controls/RadChart3D/Extensions/ExtendedCameraExtension.cs
using Telerik.Windows.Controls.Charting;

namespace OpenSilver.Samples.TelerikUI.RadChart3D.Extensions
{
    public class ExtendedCameraExtension : CameraExtension
    {
        public override void Attach(ChartArea owner)
        {
            base.Attach(owner);
            owner.MouseWheel += OnMouseWheel;
        }

        public override void Detach(ChartArea owner)
        {
            base.Detach(owner);
            owner.MouseWheel -= OnMouseWheel;
        }


        private void OnMouseWheel(object sender, System.Windows.Input.MouseWheelEventArgs e)
        {
            if (!ZoomEnabled)
            {
                return;
            }

            if (e.Delta > 0)
            {
                Zoom(1.1);
            }
            else
            {
                Zoom(0.9);
            }

            e.Handled = true;
        }
    }
}
